src/utils/arrow.rs

```diff
@@ -71,3 +74,42 @@ pub(crate) fn arrow_schema_from_column_names_and_range(
 pub(crate) fn record_batch_to_pybytes<'p>(py: Python<'p>, rb: &RecordBatch) -> Result<&'p PyBytes> {
     record_batch_to_bytes(rb).map(|bytes| PyBytes::new(py, bytes.as_slice()))
 }
+
+/// Arrow array to Python.
+pub(crate) fn to_python_array(
+    array: &ArrayRef,
+    py: Python,
+    pyarrow: &PyModule,
+) -> PyResult<PyObject> {
+    let ffi_array = ArrowArray::try_new(array.data().to_owned()).with_context(|| "haha")?;
+    let (array_ptr, schema_ptr) = ArrowArray::into_raw(ffi_array);
+    let array = pyarrow.getattr("Array")?.call_method1(
+        "_import_from_c",
+        (array_ptr as Py_uintptr_t, schema_ptr as Py_uintptr_t),
+    )?;
+    Ok(array.to_object(py))
+}
+
+/// RecordBatch to Python.
+pub(crate) fn to_python_record_batch(
+    rb: &RecordBatch,
+    py: Python,
+    pyarrow: &PyModule,
+) -> PyResult<PyObject> {
+    let mut arrays = Vec::with_capacity(rb.num_columns());
+    for array in rb.columns() {
+        let array_object = to_python_array(array, py, pyarrow)?;
+        arrays.push(array_object);
+    }
```

Comment:
maybe this can be optimized...

Reply:
Maybe `array` can be made a PyIterator ? A TODO for later is enough for me :)